dora/core/client/fs/src/main/java/alluxio/client/file/FileSystemContext.java

```diff
@@ -157,6 +158,8 @@ public class FileSystemContext implements Closeable {
   private volatile ConcurrentHashMap<ClientPoolKey, BlockWorkerClientPool>
       mBlockWorkerClientPoolMap;
 
+  private MembershipManager mMembershipManager;
```

Comment:
do we need to use `volatile` ?

Reply:
i'm assuming FileSystemCtx.initContext is called at time of creation of the filesystemctx? so by the time any req handling threads is going to use this, its already set? And MembershipManager is practically stateless, it merely uses AlluxioEtcdClient to get all infos like livemembers / allmembers etc.

Reply:
i actually have a problem with extensive usage of volatile as it only guards the access to the variable to make it visible, but not for purpose of synchronization to set of operation done to the variable such as CAS CompareAndUpdate Atomic inc/dec